src/icmp/Icmp6.cc

```diff
@@ -146,7 +148,7 @@ Icmp6::SendEcho(Ip::Address &to, int opcode, const char *payload, int len)
     icmp6_pktsize = sizeof(struct icmp6_hdr);
 
     // Fill Icmp6 ECHO data content
-    echo = (icmpEchoData *) (pkt + sizeof(icmp6_hdr));
+    echo = (icmpEchoData *) ((char *) pkt + sizeof(*icmp));
```

Comment:
Similarly:

```suggestion
    echo = reinterpret_cast<icmpEchoData*>(&pkt[sizeof(*icmp)]);
```

or, if necessary:

```suggestion
    echo = reinterpret_cast<icmpEchoData*>(reinterpret_cast<char*>(pkt) + sizeof(*icmp));
```